src/plugins/layout.ts

```diff
@@ -1,15 +1,13 @@
 import { config } from "./../config";
-import { addEvent, bind, getTimestamp, register } from "./../core";
+import { addEvent, bind, getTimestamp } from "./../core";
 import { assert, debug, isNumber, traverseNodeTree } from "./../utils";
-import * as LayoutStateProvider from "./layoutstateprovider";
-import { getNodeIndex, NodeIndex } from "./layoutstateprovider";
-import { ShadowDom } from "./shadowdom";
+import { getNodeIndex, NodeIndex, createGenericLayoutState, createLayoutState, IgnoreTag } from "./layout/stateprovider";
+import { ShadowDom } from "./layout/shadowdom";
 
 // Constants
-export const LayoutEventName = "Layout";
-const DistanceThreshold = 5;  // Pixels
-
-class Layout implements IComponent {
+export default class Layout implements IPlugin {
+  private eventName: string = "Layout";
+  private distanceThreshold: number = 5;
```

Comment:
No need for explicit typing, because you are initializing them to a specific value. Not a big deal at all, but just for consistency purposes, please either remove it or add it in other places as well :)